src/Analyzers/VisualBasic/CodeFixes/IncludeParenthesesForMethodInvocations/VisualBasicIncludeParenthesesForMethodInvocationsCodeFixProvider.vb

```diff
@@ -0,0 +1,88 @@
+        Public Sub New()
+        End Sub
+
+        Public Overrides ReadOnly Property FixableDiagnosticIds As ImmutableArray(Of String) =
+            ImmutableArray.Create(IDEDiagnosticIds.IncludeParenthesesForMethodInvocationsDiagnosticId)
+
+        Friend Overrides ReadOnly Property CodeFixCategory As CodeFixCategory = CodeFixCategory.CodeStyle
+
+        Public Overrides Async Function RegisterCodeFixesAsync(context As CodeFixContext) As Task
+            Dim root = Await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(False)
+            For Each diagnostic In context.Diagnostics
+                Dim node = root.FindNode(diagnostic.AdditionalLocations(0).SourceSpan, getInnermostNodeForTie:=True)
+                While Not (node Is Nothing OrElse node.IsKind(SyntaxKind.InvocationExpression))
+                    node = node.Parent
+                End While
+
+                If TypeOf node IsNot InvocationExpressionSyntax Then
+                    Return
+                End If
+                Dim invocationExpression = DirectCast(node, InvocationExpressionSyntax)
+                If invocationExpression.ArgumentList Is Nothing Then
+                    ' Add the parentheses.
+                    context.RegisterCodeFix(New MyCodeAction(
+                        VisualBasicAnalyzersResources.Add_method_invocation_parentheses,
+                        Function(ct) FixAsync(context.Document, diagnostic, ct)),
+                        diagnostic)
+                Else
+                    ' Remove the parentheses.
+                    context.RegisterCodeFix(New MyCodeAction(
+                        VisualBasicAnalyzersResources.Remove_method_invocation_parentheses,
+                        Function(ct) FixAsync(context.Document, diagnostic, ct)),
+                        diagnostic)
+                End If
+            Next
+        End Function
+
+        Protected Overrides Async Function FixAllAsync(
+            document As Document, diagnostics As ImmutableArray(Of Diagnostic),
+            editor As SyntaxEditor, cancellationToken As CancellationToken) As Task
+            Dim root = Await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(False)
+            For Each diagnostic In diagnostics
+                Dim node = root.FindNode(diagnostic.AdditionalLocations(0).SourceSpan, getInnermostNodeForTie:=True)
+                While Not (node Is Nothing OrElse node.IsKind(SyntaxKind.InvocationExpression))
+                    node = node.Parent
+                End While
+                If TypeOf node IsNot InvocationExpressionSyntax Then
+                    Return
+                End If
+
+                Dim invocationExpression = DirectCast(node, InvocationExpressionSyntax)
+                If invocationExpression.ArgumentList Is Nothing Then
+                    editor.ReplaceNode(invocationExpression, invocationExpression.WithArgumentList(SyntaxFactory.ArgumentList()))
```

Comment:
You might have to handle TrailingTrivia here, but probably want a testcase for it, see #45221

Reply:
@NiKiZe, I'll have this in mind to add a test case for and to handle it. But currently it doesn't even work when there aren't trivia. I have to fix that first.